src/Cookie.php

```diff
@@ -40,88 +71,120 @@ final class Cookie
     private string $value;
 
     /**
-     * @var string|null domain of the cookie
+     * @var DateTimeInterface|null RFC-1123 date at which the cookie expires.
+     * @see https://tools.ietf.org/html/rfc6265#section-4.1.1
      */
-    private ?string $domain = null;
+    private ?DateTimeInterface $expire = null;
 
     /**
-     * @var int the timestamp at which the cookie expires. This is the server timestamp.
-     * Defaults to 0, meaning "until the browser is closed".
+     * @var int|null maximum age of cookie in seconds
      */
-    private int $expire = 0;
+    private ?int $maxAge = null;
 
     /**
-     * @var string the path on the server in which the cookie will be available on. The default is '/'.
+     * @var string|null domain of the cookie.
      */
-    private string $path = '/';
+    private ?string $domain = null;
 
     /**
-     * @var bool whether cookie should be sent via secure connection
+     * @var string|null the path on the server in which the cookie will be available on.
      */
-    private bool $secure = true;
+    private ?string $path = null;
```

Comment:
I'd make `/` default since it's what majority users expect and add `withoutPath()` method for choosing not to send it in the header.

Reply:
Again, see my other answer. I just made a way to build a minimal valid cookie and am setting defaults in constructor with flag which is on by default.